cumulus/client/consensus/aura/src/collators/slot_based/collation_task.rs

```diff
@@ -140,6 +146,28 @@ async fn handle_collation_message<Block: BlockT>(
 	);
 
 	if let MaybeCompressedPoV::Compressed(ref pov) = collation.proof_of_validity {
+
+		if let Ok(relay_parent_header) = relay_client.header(BlockId::Hash(relay_parent)).await {
```

Comment:
Please adjust this a bit:
- You can pattern match directly for `Ok(Some(relay_parent_header))`
- Check for `export_pov` as outermost check, otherwise we always fetch the `relay_parent_header` here even though we don't need to (most of the time, `export_pov` will be false, it is rarely used).